xmtp/src/session.rs

```diff
@@ -0,0 +1,20 @@
+use vodozemac::olm::{OlmMessage, Session as OlmSession};
+
+pub struct Session {
+    session: OlmSession,
+}
+
+impl Session {
+    pub fn new(session: OlmSession) -> Self {
+        Self { session }
+    }
+
+    pub fn id(&self) -> String {
+        self.session.session_id()
+    }
+
+    // TODO: Replace the OlmMessage with our own message wrapper? Or leave up to the caller?
+    pub fn encrypt(&mut self, plaintext: &[u8]) -> OlmMessage {
+        self.session.encrypt(plaintext)
```

Comment:
This doesn't look like it provides any protection for unknown key-share attacks.

https://gitlab.matrix.org/matrix-org/olm/blob/master/docs/olm.md#message-authentication-concerns

Doesn't necessarily need to be solved here, however there should be a clear place where PreKey messages are validated

Reply:
Once we have invite proto messages, there's a lot more to validate. That's coming up next.